python_modules/dagster/dagster/_core/execution/plan/compute_generator.py

```diff
@@ -268,73 +268,87 @@ def validate_and_coerce_op_result_to_iterator(
             " return no results."
         )
     elif output_defs:
-        for position, output_def, element in _zip_and_iterate_op_result(
-            result, context, output_defs
-        ):
-            annotation = _get_annotation_for_output_position(position, context.op_def, output_defs)
-            if output_def.is_dynamic:
-                if not isinstance(element, list):
-                    raise DagsterInvariantViolationError(
-                        f"Error with output for {context.describe_op()}: "
-                        f"dynamic output '{output_def.name}' expected a list of "
-                        "DynamicOutput objects, but instead received instead an "
-                        f"object of type {type(element)}."
-                    )
-                for item in element:
-                    if not isinstance(item, DynamicOutput):
+        # Skip any return-specific validation and treat it like a generator op
```

Comment:
So everything in this file is more thrashy than I would have expected. Is there a way to put this in the execution engine more surgically?

Reply:
That was a consequence of using a nested conditional which caused a lot of lines to be reformatted from any length limit, I flattened the conditional and you can see the change clearly now.